test/alternator/test_manual_requests.py

```diff
@@ -278,3 +278,37 @@ def test_base64_malformed(dynamodb, test_table_b):
     assert r.status_code == 400
     r = put_item_binary_data_in_non_key(dynamodb, test_table_b, "YWJj??!!")
     assert r.status_code == 400
+
+def scan_with_binary_data_in_cond_expr(dynamodb, test_table_b, filter_expr, expr_attr_values):
+    payload ='''{
+        "TableName": "%s",
+        "FilterExpression": "%s",
+        "ExpressionAttributeValues": { %s }
+    }''' % (test_table_b.name, filter_expr, expr_attr_values)
+    req = get_signed_request(dynamodb, 'Scan', payload)
+    return requests.post(req.url, headers=req.headers, data=req.body, verify=True)
+
+# Tests the case where malformed binary data is placed as part of filter expression
+def test_base64_malformed_cond_expr(dynamodb, test_table_b):
+    # put some data
+    c_data = base64.b64encode(b"fefe").decode()
+    r = put_item_binary_data_in_non_key(dynamodb, test_table_b, c_data)
+    assert r.status_code == 200
+
+    malformed_data = "ZmVmZQ=!" # has the same length as c_data to test begins_with
+    exp_attr = '''":v": {"B": "%s"}''' % malformed_data
+
+    # note that expression "c = :v" or "c in(:v)" would fail on dynamodb but not on alternator
```

Comment:
Interesting. So maybe we should have a new xfail-ing test that demonstrates this difference? I think it's not important to fix, but maybe it would be good to know.
But not necessary now.

Reply:
yes, I was thinking, should we have xfail tests for things we don't want to fix? maybe. I think we don't deserialize here as an optimization.

Reply:
Yes, it's not critical to have this test and I don't know if we'll ever bother to fix it even if we do. So your call.